### ISSUE TITLE
EObjectDescriptionImpl.getEObjectURI() performance

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#337]
EObjectDescriptionImpl.getEObjectURI() is quite slow. we should evaluate if we can cache the uri

```
private URI eObjectURI;
	
	/** * * * @generated */
	public void setFragmentGen(String newFragment) {
		String oldFragment = fragment;
		fragment = newFragment;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, BuilderStatePackage.EOBJECT_DESCRIPTION__FRAGMENT,
					oldFragment, fragment));
	}

	public void setFragment(String newFragment) {
		setFragmentGen(newFragment);
		eObjectURI = null;
	}

	@Override
	public URI getEObjectURI() {
		EObject container = eContainer();
		if (container instanceof IResourceDescription) {
			URI resourceURI = ((IResourceDescription) container).getURI();
			if (eObjectURI == null || !eObjectURI.trimFragment().equals(resourceURI)) {
				eObjectURI = resourceURI.appendFragment(getFragment());
			}
			return eObjectURI;
		}
		return null;
	}
```

**Comment by cdietrich** [comment]
@szarnekow @svenefftinge what do you think

**Comment by szarnekow** [comment]
Looks good to me. I'd rather override the setter for the container along with the setter for the fragment and clear the cache there, too. This would probably save us from checking the URI again in the getter.

**Comment by cdietrich** [comment]
i discussed with ed my first version that was this way and he proposed this version to catch uri changes on the container

**Comment by cdietrich** [comment]
fixed in Xtext 2.13
